### PR TITLE
fix: npm links on pdf report

### PR DESCRIPTION
**Pull request by lilleeleex**

### CODE REVIEW

**Comment by lilleeleex** [comment]
Hi, even if the issue is global, here is a proposition at least on the transitive deps. I stuck to CSS styles. I'm only interested in npm stats for the moment but I imagine it's the same for gits stats? Otherwise rendering the pdf in the browser does not open a new tab, I believe that these are rules specific to browsers but I would like your opinion.

File: src/analysis/extractScannerData.js [hunk 1]
```diff
@@ -138,7 +139,7 @@ export async function buildStatsFromNsecurePayloads(payloadFiles = [], options =
             break id;
           }
 
-          stats.deps.transitive.add(`${name}@${localVersion}`);
+          stats.deps.transitive[`${name}@${localVersion}`] = { visualizerUrl: `${kNpmVisualizerUrl}/${name}@${localVersion}` };
```
Comment (PierreDemailly) [inline src/analysis/extractScannerData.js:142]:
```suggestion
          stats.deps.transitive[`${name}@${localVersion}`] = { visualizerUrl: `${kNpmVisualizerUrl}/${name}` };
```

https://www.npmjs.com/package/restore-cursor@4.0.0 <- 404
https://www.npmjs.com/package/restore-cursor <- good

Or if we want the version: https://www.npmjs.com/package/restore-cursor/v/4.0.0
```suggestion
          stats.deps.transitive[`${name}@${localVersion}`] = { visualizerUrl: `${kNpmVisualizerUrl}/${name}/v/@${localVersion}` };
```

**Review by PierreDemailly — COMMENTED** [review]
Everything else lgtm!
The css is fine and the link that doesn't open in a new tab with PDF is normal, all good 👍

**Comment by PierreDemailly** [comment]
>  I'm only interested in npm stats for the moment but I imagine it's the same for gits stats?

Indeed

**Review by fraxken — APPROVED** [review]

**Comment by lilleeleex** [comment]
Great ! It's done for all stats !

File: views/template.html [hunk 2]
```diff
@@ -58,17 +58,17 @@ <h3><span>[[=z.npm_stats.packages_count.internal]]</span>Internal Dependencies</
...
-                [[ for (const flag of Object.values(option.flags)) {]]
-                  <span class="emoji" title="[[=flag.tooltipDescription]]">
-                    [[=flag.emoji]]
-                  </span>
-                [[ } ]]
-              [[ } ]]
-            </p>
+          <li class="npm-packages-item">
+            <a href="[[=option.links.npm]]" target="_blank" rel="noopener">
+              <span class="">[[=name]]</span>
```
Comment (PierreDemailly) [inline views/template.html:63]:
```suggestion
              <span>[[=name]]</span>
```

File: views/template.html [hunk 3]
```diff
@@ -82,60 +82,64 @@ <h3><span>[[=z.npm_stats.packages_count.external]]</span>Third-party Dependencie
...
```
Comment (PierreDemailly) [inline views/template.html:87]:
```suggestion
              <span>[[=name]]</span>
```

File: views/template.html [hunk 4]
```diff
@@ -82,60 +82,64 @@ <h3><span>[[=z.npm_stats.packages_count.external]]</span>Third-party Dependencie
...
+          <h3><span>[[=Object.keys(z.npm_stats.deps.node).length]]</span>Node.js Core Dependencies</h3>
         </div>
         <ul class="node-list">
-          [[ for (const name of z.npm_stats.deps.node) { ]]
-          <li data-value="[[=name]]" tabindex="0" role="link" class="link">
-            <p>[[=name]]</p>
+          [[ for (const [name, { visualizerUrl }] of Object.entries(z.npm_stats.deps.node)) { ]]
+          <li class="node-item">
+            <a href="[[=visualizerUrl]]" target="_blank" rel="noopener">
+              <span class="">[[=name]]</span>
```
Comment (PierreDemailly) [inline views/template.html:141]:
```suggestion
              <span>[[=name]]</span>
```

File: views/template.html [hunk 5]
```diff
@@ -224,17 +228,17 @@ <h3><span>[[=z.git_stats.packages_count.internal]]</span>Internal Dependencies</
...
-                [[ for (const flag of Object.values(option.flags)) {]]
-                  <span class="emoji" title="[[=flag.tooltipDescription]]">
-                    [[=flag.emoji]]
-                  </span>
-                [[ } ]]
-              [[ } ]]
-            </p>
+          <li class="npm-packages-item">
+            <a href="[[=option.links.repository]]" target="_blank" rel="noopener">
+              <span class="">[[=name]]</span>
```
Comment (PierreDemailly) [inline views/template.html:233]:
```suggestion
              <span>[[=name]]</span>
```

File: views/template.html [hunk 6]
```diff
@@ -248,17 +252,17 @@ <h3><span>[[=z.git_stats.packages_count.external]]</span>Third-party Dependencie
...
```
Comment (PierreDemailly) [inline views/template.html:257]:
```suggestion
              <span>[[=name]]</span>
```

File: views/template.html [hunk 7]
```diff
@@ -270,38 +274,42 @@ <h3><span>[[=z.git_stats.packages_count.external]]</span>Third-party Dependencie
...
         </div>
-        <ul>
-          [[ for (const name of z.git_stats.deps.transitive) { ]]
-          <li>
-            <p>[[=name]]</p>
+        <ul class="npm-packages-list">
+          [[ for (const [name, { links }] of Object.entries(z.git_stats.deps.transitive)) { ]]
+          <li class="npm-packages-item">
+            <a href="[[=links.repository]]" target="_blank" rel="noopener">
+              <span class="">[[=name]]</span>
```
Comment (PierreDemailly) [inline views/template.html:296]:
```suggestion
              <span>[[=name]]</span>
```

File: views/template.html [hunk 8]
```diff
@@ -270,38 +274,42 @@ <h3><span>[[=z.git_stats.packages_count.external]]</span>Third-party Dependencie
...
         </div>
-        <ul>
-          [[ for (const name of z.git_stats.deps.node) { ]]
-          <li>
-            <p>[[=name]]</p>
+        <ul class="node-list">
+          [[ for (const [name, { visualizerUrl }] of Object.entries(z.git_stats.deps.node)) { ]]
+          <li class="node-item">
+            <a href="[[=visualizerUrl]]" target="_blank" rel="noopener">
+              <span class="">[[=name]]</span>
```
Comment (PierreDemailly) [inline views/template.html:311]:
```suggestion
              <span>[[=name]]</span>
```

**Review by PierreDemailly — CHANGES_REQUESTED** [review]

**Review by PierreDemailly — APPROVED** [review]
lgtm

**Comment by PierreDemailly** [comment]
@lilleeleex Is this still WIP or can we make the PR ready ?

**Comment by lilleeleex** [comment]
@PierreDemailly everthing is OK !
thx for the review

**Comment by lilleeleex** [comment]
I was wondering if there was a PR policy ? (ie: squash commits)

**Comment by PierreDemailly** [comment]
> I was wondering if there was a PR policy ? (ie: squash commits)

We do squash-merge anyway except when we wanna keep multiple commits

**Review by fraxken — APPROVED** [review]

**Comment by PierreDemailly** [comment]
@all-contributors please add @lilleeleex for code

**Pull request merged**